include/xpedite/common/WaitFreePointerEscrow.H

```diff
@@ -0,0 +1,100 @@
+//////////////////////////////////////////////////////////////////////////////////////////////////////
+//
+// WaitFreePointerEscrow - a SPSC wait free data structure for sharing a pointer between two threads.
```

Comment:
should we just called this `WaitFreeEscrow`, since it's really an escrow for a heap allocated object. 
The ownership of the object is managed and not the pointer.